public/js/utils/source-search.js

```diff
@@ -32,6 +33,31 @@ function getSearchCursor(cm, query, pos) {
     typeof query == "string" && query == query.toLowerCase());
 }
 
+function searchOverlay(query) {
+  query = query.replace(/[\-\[\]\/\{\}\(\)\*\+\?\.\\\^\$\|]/g, "\\$&");
```

Comment:
lol. this PR is crazy town, do you have some context on how you came up with it :)